src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -399,53 +379,95 @@ internal static string GetRuntimeDirectoryOpt()
 
         internal static bool TryCreateServerCore(string clientDir, string pipeName)
         {
-            // The server should be in the same directory as the client
-            string expectedPath = Path.Combine(clientDir, ServerName);
+#if NETSTANDARD1_3
+            bool isRunningOnCoreClr = CoreClrShim.IsRunningOnCoreClr;
```

Comment:
Seems like we can just rely on the runtime check.

Reply:
You would think 😢

The type `CoreClrShim` only exists in the project MSBuildTask (which is netstandard1.3), but does not exist in csc, vbc, csi, or vbi (which are multitargeted net46/netcoreapp2.0).